tests/runner/test_dataset_block.py

```diff
@@ -131,107 +169,167 @@ def test_partial_block_for_chunked_data(
 def test_partial_block_for_chunked_data_with_padding_center(
     slicing_dim: Literal[0, 1, 2]
 ):
-    block_shape = [10, 10, 10]
-    block_shape[slicing_dim] = 6
-    start_index = 3
-    data = np.ones(block_shape, dtype=np.float32)
-    global_index = [0, 0, 0]
-    global_index[slicing_dim] = 10 + start_index
+    # sizes and shapes
     global_shape_t = [10, 10, 10]
     global_shape_t[slicing_dim] = 30
     global_shape = make_3d_shape_from_shape(global_shape_t)
-    chunk_shape_t = [10, 10, 10]
-    chunk_shape_t[slicing_dim] += 4  # for padding
-    chunk_shape = make_3d_shape_from_shape(chunk_shape_t)
+
+    unpadded_chunk_shape = (10, 10, 10)
+    padding = (2, 2)
+    padded_chunk_shape_t = list(unpadded_chunk_shape)
+    padded_chunk_shape_t[slicing_dim] += padding[0] + padding[1]
+    padded_chunk_shape = make_3d_shape_from_shape(padded_chunk_shape_t)
+
+    unpadded_block_slices = 2
+    unpadded_block_shape_t = list(unpadded_chunk_shape)
+    unpadded_block_shape_t[slicing_dim] = unpadded_block_slices
+    unpadded_block_shape = make_3d_shape_from_shape(unpadded_block_shape_t)
+    padded_block_shape_t = unpadded_block_shape_t
+    padded_block_shape_t[slicing_dim] += padding[0] + padding[1]
+    padded_block_shape = make_3d_shape_from_shape(padded_block_shape_t)
+    block_start_index = 3
+
+    chunk_start_index = 10  # start of chunk within global data, without padding
+
+    data = np.ones(padded_block_shape, dtype=np.float32)
     angles = np.linspace(0, math.pi, global_shape[0], dtype=np.float32)
     block = DataSetBlock(
         data=data,
         aux_data=AuxiliaryData(angles=angles),
         slicing_dim=slicing_dim,
-        block_start=start_index,
-        chunk_start=10,
+        block_start=block_start_index,
+        chunk_start=chunk_start_index,
         global_shape=global_shape,
-        chunk_shape=chunk_shape,
-        padding=(2, 2),
+        chunk_shape=padded_chunk_shape,
+        padding=padding,
     )
 
     assert block.is_padded is True
-    assert block.padding == (2, 2)
+    assert block.padding == padding
+    assert block.chunk_index[slicing_dim] == block_start_index
+    assert block.chunk_index_unpadded[slicing_dim] == block_start_index + padding[0]
+    assert block.chunk_shape[slicing_dim] == padded_chunk_shape[slicing_dim]
+    assert block.chunk_shape_unpadded[slicing_dim] == unpadded_chunk_shape[slicing_dim]
+    assert (
+        block.global_index[slicing_dim]
+        == chunk_start_index + padding[0] + block_start_index
+    )
+    assert (
+        block.global_index_unpadded[slicing_dim]
+        == chunk_start_index + padding[0] + block_start_index + padding[0]
+    )
```

Comment:
I'm trying to understand what `block.global_index` is, now that padding has been introduced.

From the first assertion here, given that it includes one shift of `padding[0]`, naively it seems _something_ like "`block.global_index` contains padding for the chunk that the block belongs to, but not padding for the block", is that right?

And if I'm honest, I'm not 100% sure if my question makes sense, I think I may also have some confusion about padding applied to a block as far as its consequences for index shifting. Stuff like "for the global index, if you pad a block, do you need to account for the index-shift in both `block_start_index` and `chunk_start_index`" come to mind...

Any clarifying comments would be greatly appreciated :sweat_smile: (And sorry for the dumb questions!)

Reply:
The global index is the index of the block's data within the global data. The block's data includes padding. But I agree the test is confusing - because `chunk_start_index` also includes padding. That's why padding has to added in this assert, otherwise it's applied twice. This is the relation:

```python
block_start_index = block_core_start_index - padding[0]
chunk_start_index = chunk_core_start_index - padding[0]
block_global_index = chunk_core_start_index + block_core_start_index - padding[0]
```

substituting, we get:

```python
block_global_index = chunk_core_start_index + bock_start_index
                   = chunk_start_index + padding[0] + block_start_index
```

Let me change variables in the test a bit to make this more clear....